src/utils/publishToNpm.js

```diff
@@ -2,6 +2,38 @@
 
 const { runSpawn } = require('./runSpawn')
 
+async function allowNpmPublish(version) {
+  const run = runSpawn()
+
+  // We need to check if the package was already published. This can happen if
+  // the action was already executed before, but it failed in its last step
+  // (GH release).
+  let packageName = null
+  try {
+    const packageInfo = await run('npm', ['view', '--json'])
+    packageName = packageInfo ? JSON.parse(packageInfo).name : null
+  } catch (error) {
+    if (!error?.message?.match(/code E404/)) {
```

Comment:
since we're asking to return json, don't we have access to json instead of doing a regexp against the full error message?

this is what happens when running this command on the terminal on a package that doesn't exist. you see there's both console output and json, but I'm not sure if we have both in the error object here. if we do have json, maybe it's better to use that instead

```sh
$ npm view --json
npm ERR! code E404
npm ERR! 404 Not Found - GET https://registry.npmjs.org/react-patterns-workshop - Not found
npm ERR! 404
npm ERR! 404  'react-patterns-workshop@latest' is not in this registry.
npm ERR! 404 You should bug the author to publish it (or use the name yourself!)
npm ERR! 404
npm ERR! 404 Note that you can also install from a
npm ERR! 404 tarball, folder, http url, or git url.
{
  "error": {
    "code": "E404",
    "summary": "Not Found - GET https://registry.npmjs.org/react-patterns-workshop - Not found",
    "detail": "\n 'react-patterns-workshop@latest' is not in this registry.\nYou should bug the author to publish it (or use the name yourself!)\n\nNote that you can also install from a\ntarball, folder, http url, or git url."
  }
}
```

Reply:
the stderr output for the command is currently part of the error message string only - https://github.com/nearform/optic-release-automation-action/blob/main/src/utils/runSpawn.js#L25-L29